.github/workflows/internal-archive-release.yml

```diff
@@ -139,33 +207,52 @@ jobs:
         env:
           SECRETS_CONTEXT: ${{ toJson(secrets) }}
 
-      - name: "Install Spark Dependencies"
-        if: ${{ contains(github.repository, 'dbt-labs/dbt-spark') }}
-        run: |
-          sudo apt-get update
-          sudo apt-get install libsasl2-dev
 
       - name: "Set up Python - ${{ env.PYTHON_TARGET_VERSION }}"
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.PYTHON_TARGET_VERSION }}
 
-      - name: "Install python tools"
+      - name: "Install Python Dependencies"
         run: |
           python -m pip install --user --upgrade pip
           python -m pip --version
           python -m pip install tox
           tox --version
 
-      - name: "Run Tests"
+      - name: "Run Tests using tox"
         run: tox
 
+
+  run-integration-tests-hatch:
+    name: 'Integration Tests (Hatch)'
+    needs: [env-var-setup, job-setup]
+    if: "${{ needs.env-var-setup.outputs.is_hatch_adapter == 'true' }}"
+
+    uses: "dbt-labs/dbt-postgres/.github/workflows/integration-tests.yml@main"
+    with:
+      core_branch: "main"
+      dbt_adapters_branch: "main"
+
+
+  ####################
+  # Artifact Handling
+  ####################
+
   create-internal-release:
     name: Create release for internal archive
     runs-on: ubuntu-latest
-    needs: [run-unit-tests, run-integration-tests]
+    needs: [run-unit-tests, run-integration-tests, run-integration-tests-hatch]
+    # Build artifact if
+    # 1. Spark invoked (it runs tests via its in-repo workflow)
+    # 2. Integration tests passed at least via one job above
+    if: |
+      success() ||
+      inputs.dbms_name == 'spark' ||
```

Comment:
Mike suggested we make this an env var but this is impossible based on how the execution runtime works. I tried about 5 other options and none worked. It's because GHA has limited execution flow options.